### PR TITLE
add `user` support to `report` (manual report)

### PR DESCRIPTION
**Pull request by tjmehta**
I added support for adding user to the error context of the manual report method.

### CODE REVIEW

**Comment by tjmehta** [comment]
I signed it!

**Comment by cristiancavalli** [comment]
Hey there @tjmehta!
Thanks for you pull-request! I will be reviewing this asap (bandwidth is a little tight today but I should be able to get to it).

Thanks again for your contributions!

**Comment by cristiancavalli** [comment]
Hey @tjmehta after some discussion we have leaned more towards the possibility of exposing the ErrorMessage class for direct creation and consumption so as not to make invocation of manual handler too hard to parse. Please take a look at #74 and let us know what you think

**Comment by DominicKramer** [comment]
Thank you for the pull request.  However, this PR has been closed because its proposed functionality is available in the [@google-cloud/error-reporting](https://www.npmjs.com/package/@google-cloud/error-reporting) module through the [ErrorMessage](https://github.com/GoogleCloudPlatform/google-cloud-node/blob/master/packages/error-reporting/src/classes/error-message.js) class.
